atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -250,8 +303,21 @@ def resolve_signing_intent(self):
         self.compose_config.set_signing_intent(signing_intent['name'])
 
     def forward_composes(self):
-        yum_repourls = [compose_info['result_repofile'] for compose_info in self.composes_info]
-        override_build_kwarg(self.workflow, 'yum_repourls', yum_repourls)
+        repos_by_arch = {}
+        # set overrides by arch if arches are available
+        for compose_info in self.composes_info:
+            for arch in compose_info.get('arches', []):
+                repos_by_arch.setdefault(arch, [])
+                repos_by_arch[arch].append(compose_info['result_repofile'])
+
+        if repos_by_arch:
+            for arch in repos_by_arch:
+                override_build_kwarg(self.workflow, 'yum_repourls',
+                                     repos_by_arch[arch], arch)
+        # set generic overrrides otherwise
+        else:
+            yum_repourls = [compose_info['result_repofile'] for compose_info in self.composes_info]
+            override_build_kwarg(self.workflow, 'yum_repourls', yum_repourls, None)
```

Comment:
We're expecting some composes to apply to all arches, while other composes to be arch specific. This should probably be handled while iterating over `self.composes_info`.

Reply:
do you expect a single container.yaml file to have both arches and non-arches composes in it? because I'm not sure that's even possible.
if it possible and I'm misreading the docs again, I'd like to see an sample container.yaml file so I have something to work against.

Reply:
Fixed after seeing the examples in the docs.